src/frontend/js/components/Pagination/index.spec.tsx

```diff
@@ -0,0 +1,209 @@
+import { act, fireEvent, render, renderHook, screen, waitFor } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
+import { useEffect } from 'react';
+import { Pagination, usePagination } from 'components/Pagination/index';
+import { History, HistoryContext } from 'data/useHistory';
+
+jest.mock('utils/indirection/window', () => ({
+  location: {},
+  scroll: jest.fn(),
+}));
+
+describe('<Pagination/>', () => {
+  const historyPushState = jest.fn();
+  const historyReplaceState = jest.fn();
+  const makeHistoryOf: (params: any) => History = () => [
+    {
+      state: { name: '', data: {} },
+      title: '',
+      url: `/`,
+    },
+    historyPushState,
+    historyReplaceState,
+  ];
+  it('does not display pagination ( 0 pages )', async () => {
+    const pagination = renderHook(() => usePagination({ itemsPerPage: 10 }));
+    await waitFor(() => expect(pagination.result.current.maxPage).toBeUndefined());
+    render(
+      <IntlProvider locale="en">
+        <Pagination {...pagination.result.current} />
+      </IntlProvider>,
+    );
+    expect(screen.queryByTestId('pagination')).toBeNull();
+  });
+  it('displays pagination for 1 pages', async () => {
+    const pagination = renderHook(() => usePagination({ itemsPerPage: 10 }));
+    act(() => pagination.result.current.setItemsCount(10));
+    await waitFor(() => expect(pagination.result.current.maxPage).toBe(1));
+    render(
+      <IntlProvider locale="en">
+        <Pagination {...pagination.result.current} />
+      </IntlProvider>,
+    );
+    expect(screen.queryByTestId('pagination')).toBeNull();
+  });
+  it('displays pagination for 2 pages', async () => {
+    const pagination = renderHook(() => usePagination({ itemsPerPage: 10 }));
+    act(() => pagination.result.current.setItemsCount(20));
+    await waitFor(() => expect(pagination.result.current.maxPage).toBe(2));
+    render(
+      <IntlProvider locale="en">
+        <Pagination {...pagination.result.current} />
+      </IntlProvider>,
+    );
+    await screen.findByTestId('pagination');
+    await screen.findByText('Page 1');
+    await screen.findByText('Currently reading page 1');
+    await screen.findByText('2');
+    await screen.findByText('Last page 2');
```

Comment:
This approach is more verbose and I don't get what it brings more?

Reply:
It's only for reading purposes, we see the difference between a waiting a element after async event and waiting a element because it's what the test want to test.
It's juste a proposal.